spec/controllers/api/v1/beneficiaries_controller_spec.rb

```diff
@@ -0,0 +1,150 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::BeneficiariesController, type: :controller do
+  let!(:beneficiary) { create :beneficiary }
+  let(:identity_type) { create :identity_type }
+  let(:supervisor) { create(:user, :supervisor, :with_can_manage_orders_permission )}
+  let(:charity_user) { create :user, :charity }
+  let(:no_permission_user) { create :user }
+  let(:parsed_body) { JSON.parse(response.body) }
+
+  describe "GET beneficiaries" do
+
+    context 'When not logged in' do
+      it "prevents reading beneficiaries", :show_in_doc do
+        get :index
+        expect(response.status).to eq(401)
+      end
+    end
+
+    context 'When logged in as Supervisor' do
+      before { generate_and_set_token(supervisor) }
+
+      it "returns 200", :show_in_doc do
+        get :index
+        expect(response.status).to eq(200)
+      end
+
+      it 'returns all beneficiaries' do
+        5.times { FactoryBot.create :beneficiary }        
+        get :index
+        expect(parsed_body['beneficiaries'].count).to eq(Beneficiary.count)
+      end
+    end
+
+    context 'When logged in as Charity user' do
+      before { generate_and_set_token(charity_user) }
+
+      it 'returns only beneficiaries created by logged in user' do
+        3.times { FactoryBot.create :beneficiary }
+        2.times { FactoryBot.create :beneficiary, created_by: charity_user }
+        get :index
+        expect(response.status).to eq(200)
+        expect(parsed_body['beneficiaries'].count).to eq(2)
+        expect(parsed_body["beneficiaries"][0]['created_by_id']).to eq(charity_user.id)
+      end
+
+      it 'returns a single beneficiary created by logged in user' do
+        new_beneficiary = FactoryBot.create :beneficiary, created_by: charity_user
+        get :show, id: new_beneficiary.id
+        expect(response.status).to eq(200)
+        expect(parsed_body['beneficiary']['id']).to eq(new_beneficiary.id)
+        expect(parsed_body['beneficiary']['created_by_id']).to eq(charity_user.id)
+      end
+
+      it 'denies access to beneficiaries created by someone else' do
+        get :show, id: beneficiary.id
+        expect(response.status).to eq(403)
+      end
+
+    end
+
+  end
+
+  describe "POST /beneficiaries" do
+
+    context 'When not logged in' do
+      it "denies creation of a beneficiary" do
+        payload = FactoryBot.build(:beneficiary).attributes.except('id', 'updated_at', 'created_at', 'created_by_id')
```

Comment:
Quick tip to DRY up the code - you could extract `payload` here up into a `let` command at the top of the file or relevant parent `context`

`let(:payload) { FactoryBot.build()... }`

and then remove all the `payload =` lines.